x-pack/plugins/data_visualizer/public/application/index_data_visualizer/search_strategy/requests/get_boolean_field_stats.ts

```diff
@@ -80,15 +80,34 @@ export const fetchBooleanFieldsStats = (
         if (!isIKibanaSearchResponse(resp)) return resp;
 
         const aggregations = resp.rawResponse.aggregations;
-        const aggsPath = getSamplerAggregationsResponsePath(samplerShardSize);
+        const aggsPath = ['sample'];
+        const sampleCount = get(aggregations, [...aggsPath, 'doc_count'], 0);
 
         const batchStats: BooleanFieldStats[] = fields.map((field, i) => {
           const safeFieldName = field.fieldName;
+          // Sampler agg will yield doc_count that's bigger than the actual # of sampled records
+          // because it uses the stored _doc_count if available
+          // https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-doc-count-field.html
+          // therefore we need to correct it by multiplying by the sampled probability
```

Comment:
I am not sure this is doing what you are commenting.

I think this is scaling the count back DOWN via the probability (random_sampler not only uses `_doc_count`, but scales sampled counting statistics by the inverted probability). In doing this, you ensure that doc count is near the sampledCount, but this has nothing to do with _doc_count.